lib/apiservers/engine/backends/container.go

```diff
@@ -1431,3 +1473,72 @@ func copyEscapable(dst io.Writer, src io.ReadCloser, keys []byte) (written int64
 func clientFriendlyContainerName(name string) string {
 	return fmt.Sprintf("/%s", name)
 }
+
+func processVolumeParam(volString string) (volumeFields, error) {
+	volumeStrings := strings.Split(volString, ":")
+	fields := volumeFields{}
+	switch len(volumeStrings) {
+	case 1:
+		VolumeID, err := uuid.NewUUID()
+		if err != nil {
+			return volumeFields{}, nil
+		}
+		fields.VolumeID = VolumeID.String()
+		fields.VolumeDest = volumeStrings[0]
+		fields.VolumeFlags = "rw"
+	case 2:
+		fields.VolumeID = volumeStrings[0]
+		fields.VolumeDest = volumeStrings[1]
+		fields.VolumeFlags = "rw"
+	case 3:
+		fields.VolumeID = volumeStrings[0]
+		fields.VolumeDest = volumeStrings[1]
+		fields.VolumeFlags = volumeStrings[2]
+	default:
+		//NOTE: the docker cli should cover this case. This is here for posterity.
+		return volumeFields{}, fmt.Errorf("Volume bind input is invalid : -v %s", volString)
+	}
+	return fields, nil
+}
+
+func processAnonymousVolumes(h *string, volumes map[string]struct{}, client *client.PortLayer) ([]volumeFields, error) {
+	var volumeFields []volumeFields
+
+	for v := range volumes {
+		fields, err := processVolumeParam(v)
+		log.Infof("Processed Volume arguments : %#v", fields)
+		if err != nil {
+			return volumeFields, err
```

Comment:
I'm not sure what to do here.  We tried to process some args, some args *were* processed, but we have an error.  Is the semantic to continue the container `Create` with what worked and ignore the error or stop everything?  I prefer stopping everything by returning nil, err.